llvm/utils/TableGen/DAGISelMatcherGen.cpp

```diff
@@ -294,6 +294,12 @@ void MatcherGen::EmitLeafMatchCode(const TreePatternNode &N) {
     return;
   }
 
+  if (LeafRec->isSubClassOf("Operand")) {
```

Comment:
An Operand is more of a construct of the output instruction, and less so the pattern it's matching. I'm not sure it makes sense to just let any Operand pass through as an unchecked input. I would expect such a pattern to be matching timm

Reply:
`timm` depends on target register size. Things like rounding mode are groups of bits unrelated to that size. It can create problems in pattern matching,

The rounding mode could be recognized early, during building DAG, and converted into `frmarg`. In this case `frmarg` should be allowed to appear in the input part of patterns.

Of course, the problem of pattern matching can be solved by using custom lowering.

Reply:
Things that are timm should just be a consistent, fixed size since it doesn't matter. You can just pick i32 or whatever